### PR TITLE
[flutter_tools] update to package vm_service: electric boogaloo

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Implements the bootstrap and first migration described in #52179 and backs the tools current vm service with `package:vm_service`

### CODE REVIEW

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 1]
```diff
@@ -564,16 +564,11 @@ class HotRunner extends ResidentRunner {
     if (benchmarkMode) {
       final List<Future<void>> isolateNotifications = <Future<void>>[];
       for (final FlutterDevice device in flutterDevices) {
+        await device.vmService.streamListen('Isolate');
         for (final FlutterView view in device.views) {
           isolateNotifications.add(
-            view.owner.vm.vmService.onIsolateEvent
-              .then((Stream<ServiceEvent> serviceEvents) async {
-              await for (final ServiceEvent serviceEvent in serviceEvents) {
-                if (serviceEvent.owner.name.contains('_spawn')
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/run_hot.dart:572]: TODO: verify spawn check is not necessary

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_hot.dart:572]: @zanderso I've updated the logic here, the previous event notifications we were using are technically private and aren't exposed any more. However, the entire purpose was just to guesstimate when the new isolate was ready, which I can still do by listening for an extension to be registered.

**Review by bkonyi — COMMENTED** [review]
Usage of package:vm_service LGTM

**Comment by jonahwilliams** [comment]
Thanks @bkonyi !

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
@zanderso due to https://github.com/flutter/flutter/issues/53801 , the customer testing shard started failing and the only way to fix it is a new branch

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 2]
```diff
@@ -110,9 +108,10 @@ class HotRunner extends ResidentRunner {
     // TODO(cbernaschina): check that isolateId is the id of the UI isolate.
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_hot.dart:108]: hmmm. I'm going to say if it didn't need to be done in the last 3 years, it probably doesn't need to be done ever

File: packages/flutter_tools/lib/src/run_hot.dart [hunk 3]
```diff
@@ -564,16 +564,15 @@ class HotRunner extends ResidentRunner {
     if (benchmarkMode) {
       final List<Future<void>> isolateNotifications = <Future<void>>[];
       for (final FlutterDevice device in flutterDevices) {
+        try {
+          await device.vmService.streamListen('Isolate');
+        } on vm_service.RPCError {
+          // Do nothing, we're already subcribed.
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/run_hot.dart:570]: I think so, though I've mostly been copying the pattern from other places that use package:vm_service. FYI @bkonyi

[the remark this answers is not in the document]
Comment (bkonyi) [reply packages/flutter_tools/lib/src/run_hot.dart:570]: Yup, that should be it.

**Pull request merged**